nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -433,21 +479,50 @@ def _setup_dataloader_from_config(self, cfg: DictConfig):
                 drop_last=cfg.get("drop_last", False),
             )
         else:
-            dataset = TranslationDataset(
-                dataset_src=str(Path(cfg.src_file_name).expanduser()),
-                dataset_tgt=str(Path(cfg.tgt_file_name).expanduser()),
-                tokens_in_batch=cfg.tokens_in_batch,
-                clean=cfg.get("clean", False),
-                max_seq_length=cfg.get("max_seq_length", 512),
-                min_seq_length=cfg.get("min_seq_length", 1),
-                max_seq_length_diff=cfg.get("max_seq_length_diff", 512),
-                max_seq_length_ratio=cfg.get("max_seq_length_ratio", 512),
-                cache_ids=cfg.get("cache_ids", False),
-                cache_data_per_node=cfg.get("cache_data_per_node", False),
-                use_cache=cfg.get("use_cache", False),
-                reverse_lang_direction=cfg.get("reverse_lang_direction", False),
-            )
-            dataset.batchify(self.encoder_tokenizer, self.decoder_tokenizer)
+            if not self.multilingual:
+                src_file_list = [cfg.src_file_name]
+                tgt_file_list = [cfg.tgt_file_name]
+            else:
+                src_file_list = cfg.src_file_name
+                tgt_file_list = cfg.tgt_file_name
+
+            if len(src_file_list) != len(tgt_file_list):
```

Comment:
This will be true as the user is effectively passing multiple parallel datasets. That is, in the case of en->de/fr, the user needs to pass en->de files and en->fr files. Making it 2 src files and 2 tgt files.

Reply:
Ah right, my bad